platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/DiagnosticToProblemListener.java

```diff
@@ -87,8 +87,18 @@ public void report(Diagnostic<? extends JavaFileObject> diagnostic) {
     }
 
     private static ProblemId id(Diagnostic<? extends JavaFileObject> diagnostic) {
-        String idName = diagnostic.getCode().replace('.', '-');
-        return ProblemId.create(idName, mapKindToDisplayName(diagnostic.getKind()), GradleCoreProblemGroup.compilation().java());
+        String idName = diagnosticCode(diagnostic);
+        return ProblemId.create(idName, diagnosticMessage(diagnostic), GradleCoreProblemGroup.compilation().java());
+    }
+
+    private static String diagnosticCode(Diagnostic<? extends JavaFileObject> diagnostic) {
+        String message = diagnostic.getCode();
+        return message == null ? "unknown" : message;
+    }
+
+    private static String diagnosticMessage(Diagnostic<? extends JavaFileObject> diagnostic) {
+        String message = diagnostic.getMessage(Locale.getDefault());
+        return message == null ? "Unknown" : message;
```

Comment:
"unknown" should be the same case as in line 96

Reply:
That was a wrong assumption in my head that the message needs start with an upper-case char; fixed